### PR TITLE
Improve makefile and gitignore

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR adds: `bin/` directory to gitignore and `build` task to makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,13 +15,33 @@
...
 export GO111MODULE
 GOFLAGS += -mod=vendor
 export GOFLAGS
 GOPROXY ?=
 export GOPROXY
 
+NO_DOCKER ?= 0
+ifeq ($(NO_DOCKER), 1)
+  DOCKER_CMD =
+  IMAGE_BUILD_CMD = imagebuilder
```
Comment (JoelSpeed) [inline Makefile]: I don't think this is being used, can probably drop it until it's required

Comment (alexander-demichev) [reply Makefile]: I talked about it with Alberto. Sometimes we want to run it in docker, but some commands might not work as expected so we need to revisit the functionality in future

**Pull request merged**